go/pkg/client/tree_test.go

```diff
@@ -2117,6 +2117,88 @@ func TestComputeOutputsToUploadDirectories(t *testing.T) {
 	}
 }
 
+func TestComputeOutputsToUploadFileNoPermissions(t *testing.T) {
+	tests := []struct {
+		desc           string
+		input          []*inputPath
+		wd             string
+		paths          []string
+		nodeProperties map[string]*cpb.NodeProperties
+		wantResult     *repb.ActionResult
+		wantBlobs      [][]byte
+		wantCacheCalls map[string]int
+	}{
+		{
+			desc: "foo with permissions, bar without permissions",
+			input: []*inputPath{
+				{path: "wd/foo", fileContents: fooBlob, isExecutable: true},
+				{path: "bar", fileContents: barBlob},
+			},
+			paths:          []string{"foo", "bar"},
+			nodeProperties: map[string]*cpb.NodeProperties{"foo": fooProperties},
+			wd:             "wd",
+			wantBlobs:      [][]byte{fooBlob},
+			wantResult: &repb.ActionResult{
+				OutputFiles: []*repb.OutputFile{
+					&repb.OutputFile{Path: "foo", Digest: fooDgPb, IsExecutable: true, NodeProperties: command.NodePropertiesToAPI(fooProperties)},
+				},
+			},
+			wantCacheCalls: map[string]int{
+				"bar":    1,
+				"wd/foo": 1,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		root := t.TempDir()
+		if err := construct(root, tc.input); err != nil {
+			t.Fatalf("failed to construct input dir structure: %v", err)
+		}
+
+		if err := os.Chmod(filepath.Join(root, "bar"), 0100); err != nil {
```

Comment:
This is specific to the one test case. If one is sufficient, it's not worth generalizing with test tables here.